read_test.go

```diff
@@ -7,6 +7,38 @@ import (
 	"testing"
 )
 
+func TestReadVarInt(t *testing.T) {
+	//buf := []byte{127, 10, 10, 0}
+	testCases := map[int64][]byte{
+		128:   []byte{128, 1, 10, 0},
```

Comment:
I would love to see a couple of base cases such as `0` and `1` just for sanity's sake.  I also think that since the interplay between `sz` and `remain` is so important, it would be better not to throw away the return value and to verify, making sure we have some test cases where not all bytes are consumed.